skimage/color/colorconv.py

```diff
@@ -1455,3 +1460,38 @@ def _prepare_lab_array(arr):
     if shape[-1] < 3:
         raise ValueError('Input array has less than 3 color channels')
     return dtype.img_as_float(arr, force_copy=True)
+
+
+def colormap_image(img, cmap):
+    """Draw a grayscale image by using the given colormap.
+
+    Parameters
+    ----------
+    img : (M, N) array
+        The input image.
+    cmap : string or :class:`matplotlib.colors.Colormap` instance.
+        The colormap to use
+
+    Returns
+    -------
+    out: (M, N, 3) array
+        The image drawn with the specified colormap in RGB format.
+
+    Examples
+    --------
+    >>> from skimage import data, color
+    >>> from skimage.util.colormap import viridis
+    >>> out = color.colormap_image(data.camera(), viridis)
+    """
+
+    cmap = plt.get_cmap(cmap)
+    img = img_as_ubyte(img)
+    gray_colors = np.arange(0, 256, dtype=np.int)
```

Comment:
@soupault I think the values are actually irrelevant in this case, as mpl will normalize them to 0-1 in the background anyway.

@vighneshbirodkar regarding `img_as_ubyte`, you could use `np.digitize` to convert arbitrary values to binned values in {0, ..., 255}. My only concern is stretching the contrast where it would be inappropriate to do so... I might suggest using the machinery from `io.imshow`... I would also be ok with leaving this complication to a later PR. However, I think we can reasonably expect someone to trip over this when they try to colorize a float image with values outside [0, 1].